auto-instrumentation/okhttp/okhttp-3.0/library/src/main/java/io/opentelemetry/instrumentation/library/okhttp/v3_0/OkHttp3Singletons.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.library.okhttp.v3_0;
+
+import static java.util.Collections.singletonList;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.api.instrumenter.http.HttpClientResend;
+import io.opentelemetry.instrumentation.api.instrumenter.net.PeerServiceAttributesExtractor;
+import io.opentelemetry.instrumentation.okhttp.v3_0.internal.ConnectionErrorSpanInterceptor;
+import io.opentelemetry.instrumentation.okhttp.v3_0.internal.OkHttpAttributesGetter;
+import io.opentelemetry.instrumentation.okhttp.v3_0.internal.OkHttpInstrumenterFactory;
+import io.opentelemetry.instrumentation.okhttp.v3_0.internal.TracingInterceptor;
+import io.opentelemetry.javaagent.bootstrap.internal.CommonConfig;
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+
+/** Holder of singleton interceptors for adding to instrumented clients. */
+public final class OkHttp3Singletons {
+
+    private static final Instrumenter<Request, Response> INSTRUMENTER =
+            OkHttpInstrumenterFactory.create(
+                    GlobalOpenTelemetry.get(),
+                    builder ->
+                            builder.setCapturedRequestHeaders(
+                                            CommonConfig.get().getClientRequestHeaders())
```

Comment:
@breedx-splk I grabbed this class from [here](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/okhttp/okhttp-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/okhttp/v3_0/OkHttp3Singletons.java). If I understood correctly, it seems like there are some configurations in the plain java agent that come through environment vars? If that's the case, we're going to have to find an alternative to setting those in android (in case the same are needed for android), it probably could be done statically or maybe through SPIs...

Reply:
I've moved all the config params to static fields in [here](https://github.com/open-telemetry/opentelemetry-android/pull/64/files#diff-f6fc318b6de3725f7b245121d76923d026ec324a680599a524d2bf3c6a4cc1a0).